src/univariate.jl

```diff
@@ -119,31 +119,11 @@ function tabulate(data::RealVector, midpoints::R, weights::Weights=default_weigh
 end
 
 # convolve raw KDE with kernel
-# TODO: use in-place fft
 function conv(k::UnivariateKDE, dist::UnivariateDistribution)
-    # Transform to Fourier basis
-    K = length(k.density)
-    ft = rfft(k.density)
-
-    # Convolve fft with characteristic function of kernel
-    # empirical cf
-    #  = \sum_{n=1}^N e^{i*t*X_n} / N
-    #  = \sum_{k=0}^K e^{i*t*(a+k*s)} N_k / N
-    #  = e^{i*t*a} \sum_{k=0}^K e^{-2pi*i*k*(-t*s*K/2pi)/K} N_k / N
-    #  = A * fft(N_k/N)[-t*s*K/2pi + 1]
-    c = -twoπ/(step(k.x)*K)
-    for j = 0:length(ft)-1
-        ft[j+1] *= cf(dist,j*c)
-    end
-
-    dens = irfft(ft, K)
-    # fix rounding error.
-    for i = 1:K
-        dens[i] = max(0.0,dens[i])
-    end
-
-    # Invert the Fourier transform to get the KDE
-    UnivariateKDE(k.x, dens)
+    half_grid = range(step(k.x),5*std(dist),step=step(k.x))
+    grid = [-reverse(half_grid);0;half_grid]
+    density = conv(k.density, pdf.(dist,grid)) * step(k.x)
+    UnivariateKDE(k.x, density)
 end
```

Comment:
what would your recommendation be for this? given that the methods in `KernelDensity.jl` are type piracy?